Blog restoration

Floury-Fabio opened the pull request:
https://github.com/Mate2xo/Creeons_la_coop/issues/223

Floury-Fabio committed:
use index view and controller for documents list
Before this commit documents list managed by view and controller infos
This commit is first step to the restoration of blog feature

Floury-Fabio committed:
restore info views
in order to reactive blog feature

Floury-Fabio committed:
rm useless js file
this file managed tab in index infos vies

Floury-Fabio committed:
add request tests for info

Floury-Fabio committed:
rm useless var in info and document controllers

Floury-Fabio committed:
rm useless methods in infos controllers
this methods are ensured by active admin

Floury-Fabio committed:
rm useless button.
This button served to create infos.

Floury-Fabio committed:
rm useless infos requests tests

Floury-Fabio committed:
restore request show test for info

Floury-Fabio committed:
rn useless infos views

Floury-Fabio committed:
add antichronological order to infos index

Floury-Fabio committed:
rm useless permit params method from infos controllers
The infos mangaement is ensured by active admin controllers

Floury-Fabio committed:
add enumerize gem

Floury-Fabio committed:
add category enum to infos model

Floury-Fabio committed:
add enum category in admin infos views

Floury-Fabio committed:
add i18n keys for infos attributes

Floury-Fabio committed:
add tabs in infos index
The user can now display the infos by categories

Floury-Fabio committed:
fix infos controllers test
this tests was breaked by new i18n key in config model file

Floury-Fabio committed:
add test for infos helpers #info_of_this_category

Floury-Fabio committed:
add rights at members of redactor group
We need a group which manage the blog section
For that we add an access at active admin info panel

Floury-Fabio committed:
add task which remove old infos
More precisely: removes the infos whose the last update dateback to more than one month

Floury-Fabio committed:
add a policy/active_admin namespace
We must allow redactors to access to active/infos
and denie the access to others active admin views

Floury-Fabio committed:
add a localization of a list of specific groupname
The groups with a name includes in this list grant rights to members included it

Floury-Fabio committed:
add test for active_admin policies

Floury-Fabio committed:
add published to info

Floury-Fabio committed:
add published in active admin infos views

Floury-Fabio committed:
upgrade infos helpers

Floury-Fabio committed:
enable access to unpublished infos for visitor

Floury-Fabio committed:
add doc in infos factory

Floury-Fabio committed:
correct spelling errors

Mate2xo commented on app/models/group.rb:
```diff
@@ -16,4 +16,11 @@ class Group < ApplicationRecord
+    return nil unless list_of_groups_with_rights.include?(name_to_localize)
+
+    I18n.t("activerecord.attributes.group.localized_names.#{name_to_localize}")
+  end
```
If I understand correctly, in the end this method is intended to be used only with the argument `'redactor'`.
Why not use `Group.human_enum_name('redactor')` then?

Mate2xo commented on app/models/info.rb:
```diff
@@ -6,13 +6,20 @@
 class Info < ApplicationRecord
-  validates :title, presence: true
+  extend Enumerize
+  extend ActiveModel::Naming
```
Why do you need this ?
It should be available by default, as `Info` already is an `ActiveModel`. Is it because you saw it in the Enumerize documentation? If that's the case, this option is needed only for a plain Ruby object, not a Rails object

Floury-Fabio replied:
It is a stain of one try that i forgot to remove.
I remove them.

Mate2xo commented on app/models/member.rb:
```diff
@@ -86,6 +86,13 @@ def monthly_worked_hours(date)
 
+  def redactor?
+    groups.each do |group|
+      return true if group.name == Group.localized_name_of_group_which_grant_rights_to_members('redactor')
```
Same as above, why not use `#human_enum_name`?

Mate2xo commented on db/schema.rb:
```diff
@@ -239,6 +241,16 @@
     t.datetime "updated_at", null: false
   end
 
+  create_table "temp_enrollments_from_csv", id: false, force: :cascade do |t|
+    t.bigint "member_id"
+    t.bigint "mission_id"
+    t.bigint "id"
+    t.time "old_start_time"
+    t.time "old_end_time"
+    t.datetime "start_time"
+    t.datetime "end_time"
+  end
+
```
I guess this table comes from another ticket, does it?

Floury-Fabio replied:
yes it come from https://github.com/Mate2xo/Creeons_la_coop/tree/restore_enrollments_with_csv
it is removed.

Mate2xo commented on lib/tasks/remove_old_infos.rake:
```diff
@@ -0,0 +1,9 @@
+
+# This namespace contains all tasks related to the infojmodel
+namespace :info do # rubocop:disable Metrics/BlockLength
+  desc 'removes the infos whose the last update dateback to more than one month'
```
`'deletes Info records that are more than one month old'`

Mate2xo commented on spec/factories/members.rb:
```diff
@@ -47,5 +51,12 @@
+
+    after :create do |member, options|
+      if options.redactor?
+        group = create :group, name: Group.localized_name_of_group_which_grant_rights_to_members('redactor')
```
Same as above, maybe `#human_enum_name`?

Mate2xo commented on spec/features/document_uploads_at_info_section_spec.rb:
```diff
@@ -26,17 +26,17 @@
   context "when an admin uploads a document" do
```
I think we could rename this file as `documents_uploads_at_document_page_spec.rb`, since the `Info` and `Document` now have their own dedicated pages

Floury-Fabio committed:
remove useless table form schema
this table come from https://github.com/Mate2xo/Creeons_la_coop/tree/restore_enrollments_with_csv

Floury-Fabio committed:
remove useless extend from info model

Floury-Fabio committed:
correct spelling errors

Floury-Fabio committed:
rename document_uploads_at_document_info_section_spec.rb
in document_uploads_at_document_page_section_spec.rb

Floury-Fabio committed:
add enum multiple for group rights management
